modules/appeals_api/app/models/appeals_api/higher_level_review.rb

```diff
@@ -303,40 +255,36 @@ def add_error_informal_conference_rep_will_not_fit_on_form
 
     # validation (header)
     def birth_date_is_a_date
-      add_error("Veteran birth date isn't a date: #{birth_date_string.inspect}") unless birth_date
+      add_error("Veteran birth date isn't a date: #{birth_date.raw_date.inspect}") unless birth_date.valid?
     end
 
     # validation (header)
     def birth_date_is_in_the_past
-      return unless birth_date
+      return unless birth_date.valid?
 
-      add_error("Veteran birth date isn't in the past: #{birth_date}") unless self.class.past? birth_date
+      add_error("Veteran birth date isn't in the past: #{birth_date.raw_date}") unless birth_date.in_the_past?
     end
 
     # validation
     def contestable_issue_dates_are_valid_dates
-      return unless contestable_issues
+      return if contestable_issues.blank?
 
-      contestable_issues.each_with_index do |ci, index|
-        decision_date_is_valid(ci&.dig('attributes', 'decisionDate').to_s, index)
+      contestable_issues.each_with_index do |issue, index|
+        decision_date_invalid(issue, index)
+        decision_date_not_in_past(issue, index)
```

Comment:
I think this is an improvement over what we had! 👍

I still think we could probably make it clearer what the `decision_date_invalid` and `decision_date_not_in_past` methods do though (from a naming perspective). YMMV but I would expect `*_invalid` and `*_not_in_past` type methods to have a truthy or falsey return value and not have side effects (of manually adding errors in this case). Basically, I have to go read their definitions to see that they do.

Reply:
I agree. I'm not sold on this implementation, I was already creating the contestable_issue class, so wanted to clean it up somewhat. Further improvements coming!